From 4639a92dc56793b596f3a9ae0ec1a60d67a617f5 Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Tue, 18 Feb 2020 21:10:22 +0000
Subject: [PATCH 2/2] powerpc: config: defang gcc check for stack-protector
 support

When we rebase kernel configs in Alpine, it is common to do so from
an environment where CARCH does not match CHOST.

Accordingly, we assume that the gcc check will always pass, and
simply remove it in preference for always enabling the
HAVE_STACKPROTECTOR config option.

Signed-off-by: Ariadne Conill <ariadne@dereferenced.org>
---
 arch/powerpc/Kconfig | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/powerpc/Kconfig b/arch/powerpc/Kconfig
index 3e56c9c2f16e..f0b7d77aaef1 100644
--- a/arch/powerpc/Kconfig
+++ b/arch/powerpc/Kconfig
@@ -181,8 +181,7 @@ config PPC
 	select HAVE_ASM_MODVERSIONS
 	select HAVE_C_RECORDMCOUNT
 	select HAVE_CBPF_JIT			if !PPC64
-	select HAVE_STACKPROTECTOR		if PPC64 && $(cc-option,-mstack-protector-guard=tls -mstack-protector-guard-reg=r13)
-	select HAVE_STACKPROTECTOR		if PPC32 && $(cc-option,-mstack-protector-guard=tls -mstack-protector-guard-reg=r2)
+	select HAVE_STACKPROTECTOR
 	select HAVE_CONTEXT_TRACKING		if PPC64
 	select HAVE_COPY_THREAD_TLS
 	select HAVE_DEBUG_KMEMLEAK
-- 
2.24.0

